azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/toolkit/lib/auth/AzureAccount.java

```diff
@@ -45,128 +47,144 @@ public Account account() throws AzureToolkitAuthenticationException {
-        account = targetAccount;
-        if (account.checkAvailable()) {
-            if (!account.isAuthenticated()) {
-                account.authenticate().block();
-            }
-        } else {
-            if (account.entity.getLastError() != null) {
-                throw new LoginFailureException(account.entity.getLastError().getMessage(), account.entity.getLastError());
-            } else {
-                throw new LoginFailureException(String.format("Cannot get credential from auth method '%s'.", targetAccount.getMethod()));
-            }
-        }
-        return this;
+    public Account account(AuthType type) {
+        Account ac = buildAccountMap().get(type).get();
+        ac.checkAvailable();
+        return ac;
+    }
+
+    public AzureAccount login(@Nonnull AuthType type) {
+        return blockMonoAndReturnThis(loginAsync(type));
+    }
+
+    public AzureAccount login(@Nonnull Account targetAccount) {
+        return blockMonoAndReturnThis(loginAsync(targetAccount));
+    }
+
+    public AzureAccount login(@Nonnull AuthConfiguration auth) {
+        return blockMonoAndReturnThis(loginAsync(auth));
     }
 
-    public AzureAccount logout() {
+    public void logout() {
         if (this.account != null) {
             this.account.logout();
             this.account = null;
         }
-        return this;
     }
 
     public boolean isAuthenticated() {
         return this.account != null && this.account.isAvailable() && this.account.isAuthenticated();
     }
 
-    public AzureAccount login(AuthType type) throws LoginFailureException {
-        AuthConfiguration auth = new AuthConfiguration();
-        auth.setType(type);
-        loginWithAuthConfiguration(auth);
-        return this;
+    public Mono<Account> authenticateAccounts(List<Account> accounts) {
+        Mono<Account> current = accounts.get(0).authenticate();
+        for (int i = 1; i < accounts.size(); i++) {
+            final Account ac = accounts.get(i);
+            current = current.onErrorResume(e -> ac.authenticate());
+        }
+        return current;
     }
 
-    public void login(@Nonnull AuthConfiguration auth) throws LoginFailureException {
-        // update the env state of AzureAccount when auth configuration has a strong configuration of env
-        Objects.requireNonNull(auth, "Null auth configuration is illegal for login.");
-        loginWithAuthConfiguration(auth);
-        if (auth.getEnvironment() != null && this.account.getEnvironment() != null
-                && this.account.getEnvironment() != auth.getEnvironment()
-                && this.account.isAuthenticated()) {
+    public Mono<Account> loginAsync(@Nonnull AuthConfiguration auth) {
+        // update the env state of AzureAccount when auth configuration has an explicit configuration of env
+        Objects.requireNonNull(auth, "Auth configuration is required for login.");
```

Comment:
Do we need to validate `auth` again as we have marked it as `Nonnull`

Reply:
Custom error message is also helpful